quantinuum-hugr/src/hugr/validate.rs

```diff
@@ -419,10 +422,19 @@ impl<'a, 'b> ValidationContext<'a, 'b> {
 
         let is_static = match from_optype.port_kind(from_offset).unwrap() {
             EdgeKind::Static(typ) => {
-                if !(OpTag::Const.is_superset(from_optype.tag())
-                    || OpTag::Function.is_superset(from_optype.tag()))
-                {
-                    return Err(InterGraphEdgeError::InvalidConstSrc {
+                if !OpTag::Const.is_superset(from_optype.tag()) {
```

Comment:
TBH I think this check is more of an assertion about the spec, than something we really need to check at runtime; there should be no other way to construct nodes with such out-ports! So happy to remove both these checks and the relevant error variants